### ISSUE TITLE
Copying objects within DataProxy

### ISSUE THREAD

**Issue by appy-one** [appy-one/acebase-core#10]
When assigning a proxied object to another location within the DataProxy, making changes to it will also change the original in memory. It updates the db at the right location though, so there is a discrepancy between in-memory and db states.

Example:
```js
const proxy = await db.ref('cars').proxy();
const cars = proxy.value;
// Copy sportscar to create a new racecar:
cars.racecar = products.sportscar;
// Make change to racecar:
cars.racecar.horsepower = 980;
if (cars.sportscar.horsepower === cars.racecar.horsepower) {
   // Not good: change was also made in original object
}
```
Note that this is normal javascript object behavior but should be prevented in this proxied database context.
The solution would be to clone the object at assignment.

**Comment by appy-one** [comment]
When cloning assigned objects, this will also fix unexpected situations where the database is not updated, but the in-memory value is:
```js
const sportscar = { make: 'Ferrari', model: 'Testarossa', horsepower: 390 };
const proxy = await db.ref('cars').proxy();
const cars = proxy.value;
// Create sportscar:
cars.sportscar = sportscar;

// .. Later (other tick): ..
// Won't update database:
sportscar.top_speed = 290; // (km/h)
cars.sportscar.top_speed === 290; // true, but not in db

// Updates database and original object:
cars.sportscar.top_speed = 290;
sportscar.top_speed === 290; // true
```

When cloning the value, this will become the behavior:
```js
// Won't update database:
sportscar.top_speed = 290;
cars.sportscar.top_speed === 290; // false

// Updates database but not the original object:
cars.sportscar.top_speed = 290;
sportscar.top_speed === 290; // false
```

**Comment by appy-one** [comment]
Fix published in `acebase-core` v1.5.0
